### PR TITLE
Refine test utilities

### PR DESCRIPTION
**Pull request by ddddxxx**
## CXTestUtility

- remove `TestPublisher`, `TestLogging`, `TestObject`, `TestResources`, `TracingSubscriberEvent` and `DemandState`.
- remove `makeTestSubscriber` and other ad-hoc functions.
- rename `TestSubject` to `TracingSubject` (prepare for migration to CXTest).
- Add `DispatchQueue.concurrentPerform` and `Publisher.subscribeTracingSubscriber`.

## CXTest

- Remove `onDeinit` from `TracingSubscriber` and `TracingSubscription` as it's irrelevant to subscribe.
- Add internal `TransparentPublisher` and expose an initializer from `AnyPublisher`.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -134,9 +134,8 @@ let shimTarget = package.targets.first(where: { $0.name == "CXShim" })!
 shimTarget.dependencies = combineImp.shimTargetDependencies
 shimTarget.swiftSettings.append(contentsOf: combineImp.swiftSettings)
 
-for target in package.targets where target.isTest || target.name == "CXTestUtility" || target.name == "CXTest" {
-    target.swiftSettings.append(contentsOf: combineImp.swiftSettings)
-}
+let testUtilityTarget = package.targets.first(where: { $0.name == "CXTestUtility" })!
+testUtilityTarget.swiftSettings.append(contentsOf: combineImp.swiftSettings)
```
Comment (ddddxxx) [inline Package.swift:138]: CXTest should (and already) be ignorant about concrete combine implementation.

File: Sources/CXTest/TransparentPublisher.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import CXShim
+
+extension AnyPublisher {
+    
+    public init(_ receiveSubscriber: @escaping (AnySubscriber<Output, Failure>) -> Void) {
```
Comment (ddddxxx) [inline Sources/CXTest/TransparentPublisher.swift:5]: I believe this suit for AnyPublisher very well. So we don't need to expose TransparentPublisher.

File: Sources/CXTestUtility/TracingSubject.swift [hunk 3]
```diff
@@ -106,20 +103,25 @@ public class TestSubject<Output, Failure: Error>: Subject, TestLogging {
...
         typealias Sub = AnySubscriber<Output, Failure>
         
         let lock = Lock(recursive: true)
-        
-        public var name: String?
        
         var pub: Pub?
         var sub: Sub?
         
+        enum DemandState: Equatable {
```
Comment (ddddxxx) [inline Sources/CXTestUtility/TracingSubject.swift:120]: `DemandState` only makes sense in `TracingSubject`. I can't find a general use case so I just make it private.

File: Sources/CXTestUtility/TracingSubject.swift [hunk 4]
```diff
@@ -166,10 +168,9 @@ extension TestSubject {
             let more = sub.receive(value)
             
             self._demandRecords.withLockMutating { $0.append((.sync, more)) }
-            self.trace("sync more", more)
```
Comment (ddddxxx) [inline Sources/CXTestUtility/TracingSubject.swift:169]: I don't think we need tracing in automatic test.

File: Tests/CXFoundationTests/CoderSpec.swift [hunk 5]
```diff
@@ -8,10 +8,6 @@ class CoderSpec: QuickSpec {
     
     override func spec() {
         
-        afterEach {
-            TestResources.release()
```
Comment (ddddxxx) [inline Tests/CXFoundationTests/CoderSpec.swift:12]: We don't share test resources across test cases, do we?.

File: Sources/CXTestUtility/Extensions/Int+loop.swift [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+            body(i)
+        }
+    }
+    
+    public func times(_ body: () -> Void) {
+        self.times { _ in
+            body()
+        }
+    }
+}
```
Comment (luoxiu) [inline Sources/CXTestUtility/Extensions/Int+loop.swift:17]: ❤️ ruby

**Review by luoxiu — APPROVED** [review]
Large-scale refinement! Generally LGTM, : )

**Pull request merged**